docs/rules/prefer-arrow-callback.md

```diff
@@ -51,6 +53,23 @@ Examples of **correct** code for the `{ "allowNamedFunctions": true }` option:
 foo(function bar() {});
 ```
 
+### allowUnboundThis
+
+This is a `boolean` option and it is `true` by default. When set to `false`, this option allows the use of `this` without restriction and checks for dynamically assigned `this` values such as when using `Array.prototype.map` with a `context` argument. Normally, the rule will flag the use of `this` whenever a function does not use `bind()` to specify the value of `this` constantly.
+
+Examples of **correct** code for the `{ "allowUnboundThis": true }` option:
+
+```js
+/*eslint prefer-arrow-callback: [2, { allowUnboundThis: false }] */
```

Comment:
Minor nits the eslint comments are now using:
* string "error" instead of number 2 severity
* quoted property keys in object options